rslabel_package/rslabel/ret.py

```diff
@@ -97,12 +101,49 @@ def app(args):
                     break
             src_dir = 'in_progress/validation/'
             dest_dir = 'done/' if complete else 'unvalidated/'
+        elif tar_name in clarification_tars:
+            delete = True
+            complete = True
+
+            dir_name = os.path.dirname(args.annotations)
+
+            # Looping over to see if there are bad images, and see if the tar is complete
+            for annotation in json_contents:
+                try:
+                    ann = annotation['status']
+                    if ann == 'Bad':
+                        os.remove("{}/{}".format(dir_name, annotation['filename']))
+                except:
+                    complete = False
+                    pass
+            # If we need to delete json file, because it has bad images
+            if delete:
+                annotations = []
+                tar_base = os.path.splitext(tar_name)[0]
+                json_name = tar_base + '.json'
+                if dir_name == '':
+                    dir_name = "."
+                for f in sorted(glob.glob('{}/*.jpg'.format(tar_base))):
+                    annotations.append({'annotations': [],
+                                        'class': 'image',
+                                        'filename': os.path.basename(f),
+                                        'unlabeled': True})
+
+                os.remove(args.annotations)
+                os.remove("{}/{}".format(dir_name, dir_name + "-original.json"))
+                with open(args.annotations, 'w') as f:
+                    json.dump(annotations, f, indent=4)
+                with tarfile.open(tar_name, "w") as tar:
+                    tar.add(dir_name)
```

Comment:
We shouldn't need to modify the TAR archive and re-upload it. It's fine to leave the bad, unclear images in the archive, but just delete the annotation from the annotation file. That way, sloth will ignore it for labeling and we don't need to upload the tar archive to the server (and thus use up time).

Reply:
Most of tar files on average are between 20 Mb to 100 Mb. Even with slowest possible internet that takes around 1-2 minutes to upload. If the tar has lots of images we delete them reducing size of it.

Reply:
It is not 1-2 hours, the deletion of images takes fast, just like validation.

Reply:
As a result of this change, you will be required to upload all images from the bag twice. Once when you originally unbag and then again after clarification. That's multiple gigabytes that will have to be doubly uploaded from every pool test. It's unnecessary and will cause huge slowdowns

Reply:
Images needs to be deleted, if we do not do it locally. Then we need to delete them through sftp, which means either have connection and in loop delete it, or make a script on server that upon upload of json with required deleted images will untar and delete them, and then re-tar. In first case if somehow program is interrupted by anything as accidental Ctrl+C press, program crash or OS crash, that means we will have not finished process on the server!

Local way is more secure that ensures it will be finished even with interruption.

Reply:
If we have poor internet connection during competition we are screwed anyway, as we need to pull old datasets which are 8-20Gb of size to check, re-check, extract data, etc. Few Mb of data upload/download won't affect it as much.